### PR TITLE
Update coq-quickchick.dev

### PR DESCRIPTION
**Pull request by Lysxia**

### CODE REVIEW

**Comment by palmskog** [comment]
Looks like not a single build succeeded, and there is this error:
```
# Error: There is no dune-project file in the current directory, please add one
# with a (name <name>) field in it.
# Hint: 'dune subst' must be executed from the root of the project.
```
Any ideas what's going on?

**Comment by Lysxia** [comment]
It looks like it failed at fetching the source of coq(ide-server) but still tried to go on.

**Comment by Lysxia** [comment]
There's an earlier error message in the logs:

```
+ /usr/bin/git "push" "/builds/coq/opam/opam-root-4.11.2-2.1.2-sandbox/download-cache/git" "+refs/remotes/opam-ref-master:refs/remotes/fb9e53fc5266f3285fd2364f0292dd62" (CWD=/builds/coq/opam/opam-root-4.11.2-2.1.2-sandbox/4.11.2/.opam-switch/sources/coq-core.dev)
- remote: error: cannot lock ref 'refs/remotes/fb9e53fc5266f3285fd2364f0292dd62': reference already exists        
- To /builds/coq/opam/opam-root-4.11.2-2.1.2-sandbox/download-cache/git
-  ! [remote rejected]       opam-ref-master -> fb9e53fc5266f3285fd2364f0292dd62 (failed to update ref)
- error: failed to push some refs to '/builds/coq/opam/opam-root-4.11.2-2.1.2-sandbox/download-cache/git'
- remote: error: cannot lock ref 'refs/remotes/fb9e53fc5266f3285fd2364f0292dd62': reference already exists        
- To /builds/coq/opam/opam-root-4.11.2-2.1.2-sandbox/download-cache/git
-  ! [remote rejected]       opam-ref-master -> fb9e53fc5266f3285fd2364f0292dd62 (failed to update ref)
- error: failed to push some refs to '/builds/coq/opam/opam-root-4.11.2-2.1.2-sandbox/download-cache/git'
```

**Comment by palmskog** [comment]
@Lysxia I don't have a solution to this, but on Zulip there was a suggestion to add a ` (dirs :standard \ opam)` stanza into the project's Dune configuration. Maybe you could give it a try?

For more on this: https://dune.readthedocs.io/en/latest/reference/dune/dirs.html

**Comment by Lysxia** [comment]
I can try but it looks like the error is higher up, opam doesn't even reach the point where it's trying to build coq-quickchick.

**Comment by palmskog** [comment]
There seems to be a fundamental problem with the combination Dune+opam+Git that has shown up recently. Another thing that might "work" (or at least give us some information) is to limit Dune to `< 3.14`, since this stuff showed up recently.

File: extra-dev/packages/coq-quickchick/coq-quickchick.dev/opam [hunk 1]
```diff
@@ -8,13 +8,12 @@ bug-reports: "https://github.com/QuickChick/QuickChick/issues"
 license: "MIT"
 
 build: [
-  [make "compat"]
   ["dune" "subst"] {pinned}
```
Comment (palmskog) [inline extra-dev/packages/coq-quickchick/coq-quickchick.dev/opam:11]: It probably won't help with current problems, but I think this is the preferred idiom:
```suggestion
  ["dune" "subst"] {dev}
```

**Comment by Lysxia** [comment]
I'm happy if this passes. Feel free to do whatever with the other draft PR if you want to experiment more.

**Comment by Lysxia** [comment]
Is this a spurious failure?

**Pull request merged**